trimesh/exchange/obj.py

```diff
@@ -0,0 +1,535 @@
+            # reshape to columns
+            array = array.reshape((-1, columns))
+            # faces are going to be the first value
+            index = np.arange(3) * int(columns / 3.0)
+            # slice the faces out of the blob array
+            faces = array[:, index]
+
+            # TODO: probably need to support 8 and 12 columns for quads
+            # or do something more general
+            faces_tex, normal_idx = None, None
+            if columns == 6:
+                # if we have two values per vertex the second
+                # one is index of texture coordinate (`vt`)
+
+                # count how many delimiters are in the first face line
+                # to see if our second value is texture or normals
+                count = face_lines[0].count('/')
+                if count == columns:
+                    # case where each face line looks like:
+                    # ' 75//139 76//141 77//141'
+                    # which is vertex/nothing/normal
+                    normal_idx = array[:, index + 1]
+                elif count == int(columns / 2):
+                    # case where each face line looks like:
+                    # '75/139 76/141 77/141'
+                    # which is vertex/texture
+                    faces_tex = array[:, index + 1]
+                else:
+                    log.warning('face lines are weird: {}'.format(
+                        face_lines[0]))
+            elif columns == 9:
+                # if we have three values per vertex
+                # second value is always texture
+                faces_tex = array[:, index + 1]
+                # third value is reference to vertex normal (`vn`)
+                normal_idx = array[:, index + 2]
+        else:
+            # if we had something annoying like mixed in quads
+            # or faces that differ per-line we have to loop
+            log.warning(
+                'faces are mixed tri/quad/*, try to not do this!')
+            # TODO: allow fallback, and find a mesh we can test it on
+            faces, faces_tex, normal_idx = _parse_faces(face_lines)
+
+        # TODO: where we should have usable face normals
+        vertex_normals = None
+        # if v is not None and vn is not None and len(v) == len(vn):
+        #    vertex_normals =
+
+        # try to get usable texture
+        visual = None
+        if faces_tex is not None:
+            # convert faces referencing vertices and
+            # faces referencing vertex texture to new faces
+            # where each face
+            new_faces, mask_v, mask_vt = unmerge_faces(
+                faces=faces, faces_tex=faces_tex)
+
+            # we should NOT have messed up the faces
+            # note: this is EXTREMELY slow due to the numerous
+            # float comparisons so only use in unit tests
+            if tol.strict:
+                assert np.allclose(v[faces], v[mask_v][new_faces])
+
+            try:
+                visual = TextureVisuals(
```

Comment:
Could this functionality be added outside the `if faces_tex is not None` block? The UVs might be rubbish, but the materials are still useful. (Demo)[https://gist.github.com/jackd/ca9d4ebf39f682dea43f9979d1470a9b] showing an example that gets much nicer with the below change. I know the demo code is a disgusting dirty hack and a proper implementation of non-image based renderings would likely be preferable but I really don't have the expertise for that, and the loader would still need to be modified to save the relevant material data.

```
        visual = None
        mesh = kwargs.copy()
        if faces_tex is not None:
            # convert faces referencing vertices and
            # faces referencing vertex texture to new faces
            # where each face
            new_faces, mask_v, mask_vt = unmerge_faces(
                faces=faces, faces_tex=faces_tex)

            # we should NOT have messed up the faces
            # note: this is EXTREMELY slow due to the numerous
            # float comparisons so only use in unit tests
            if tol.strict:
                assert np.allclose(v[faces], v[mask_v][new_faces])
            uv = vt[mask_vt]
            mesh['vertices'] = v[mask_v]
            mesh['faces'] = new_faces
        else:
            uv = np.zeros((v.shape[0], 2))
            mesh['vertices'] = v
            mesh['faces'] = faces

        if material in materials:
            visual = TextureVisuals(uv=uv, material=materials[material])
        else:
            log.warning('visual creation failed for submesh!', exc_info=True)
        mesh['visual'] = visual
        mesh['vertex_normals'] = vertex_normals
        geometry[util.unique_id()] = mesh

    if len(geometry) == 1:
        ....
```

Reply:
Oh yeah, good point. There were a couple issues with people asking for that (materials with no UV's but uses the ambient/diffuse color properties).